src/esthri/src/ops/download.rs

```diff
@@ -538,7 +539,23 @@ where
     let key = format!("{}", Path::new(s3_prefix).join(s3_suffix).display());
     let dest_path = format!("{}", dest_path.display());
 
-    download(s3, bucket, &key, &dest_path).await?;
+    let file_is_compressed = dest_path.ends_with(".gz");
+
+    if decompress && file_is_compressed {
+        #[cfg(feature = "compression")]
+        {
+            let dest_path = dest_path
+                .strip_suffix(".gz")
+                .expect("Should have a gz suffix");
```

Comment:
Maybe this would be more appropriate as an Err since this returns a Result.

Reply:
There might be a nicer way to do what I'm trying to achieve here but this block will only ever be entered if the suffix is `.gz` (see the check on line 542) so this line should never error. I was thinking that having this as an expect rather than an Err would be more illustrative of this, since it isn't expected to fail.

Reply:
That sounds reasonable to me